cryptoconditions/crypto.py

```diff
@@ -1,269 +1,121 @@
         """
         Verify if the signature signs the data with this verifying key
 
         Args:
-             data (bytes|str): data to be signed
+             data (bytes|str): data verify
              signature (bytes|str): {base64|base32|base16|hex|bytes} signature to be verified
-             prefix: see super
              encoding: {base64|base32|base16|hex|bytes} encoding of the signature
         """
         try:
-            if not isinstance(data, bytes):
-                data = data.encode('ascii')
-            if encoding == 'base58':
-                super().verify(base58.b58decode(signature), data, prefix=prefix)
-            else:
-                super().verify(signature, data, prefix=prefix, encoding=encoding)
-        except ed25519.BadSignatureError:
+            # The reason for using raw_signatures here is because the verify method of pynacl expects the message
+            # and the signature to have the same encoding. Basically pynacl does:
+            #   encoder.decode(signature + message)
+            raw_signature = _get_nacl_encoder(encoding).decode(signature)
+            super().verify(data, raw_signature)
+        except nacl.exceptions.BadSignatureError:
             return False
 
         return True
 
-    def to_ascii(self, prefix="", encoding='base58'):
-        """
-        convert external value to ascii with specified encoding
-
-        Args:
-            prefix (str):
-            encoding (str): {'base58'|'base64'|'base32'|'base16'|'hex'}
-
-        Returns:
-            bytes: encoded string
-        """
-        if encoding == 'base58':
-            return base58.b58encode(self.vk_s).encode('ascii').decode('ascii').rstrip("=").encode()
-        else:
-            return super().to_ascii(prefix=prefix, encoding=encoding)
-
-    @staticmethod
-    def encode(public_base64):
-        """
-        Encode the public key represented by base64 to base58
-
-        Args:
-            public_base64
-        """
-        return Ed25519SigningKey.encode(public_base64)
-
-    @staticmethod
-    def decode(public_base58):
-        """
-        Decode the base58 public_value to base64
-
-        Args:
-            public_base58
-        """
-        return Ed25519SigningKey.decode(public_base58)
+    def encode(self, encoding='base58'):
+        return super().encode(encoder=_get_nacl_encoder(encoding))
 
 
 def ed25519_generate_key_pair():
     """
     Generate a new key pair and return the pair encoded in base58
     """
-    sk, vk = ed25519.create_keypair()
+    sk = nacl.signing.SigningKey.generate()
     # Private key
-    private_value_base58 = Ed25519SigningKey(base58.b58encode(sk.to_bytes())).to_ascii()
+    private_value_base58 = sk.encode(encoder=Base58Encoder)
 
     # Public key
-    public_value_compressed_base58 = Ed25519VerifyingKey(base58.b58encode(vk.to_bytes())).to_ascii()
+    public_value_compressed_base58 = sk.verify_key.encode(encoder=Base58Encoder)
```

Comment:
In regards to #23, we would want this to be `.decode()`ed before it gets returned. This seems to only be used outside of the library, and so decoding it into a string is easier for others to use. A named tuple would also be nice here 😉.